opm/simulators/flow/FIBlackoilModel.hpp

```diff
@@ -59,23 +63,39 @@ class FIBlackOilModel : public BlackOilModel<TypeTag>
     FIBlackOilModel(Simulator& simulator)
         : BlackOilModel<TypeTag>(simulator)
     {
+#ifdef _OPENMP
+        const auto& gv = this->gridView_;
+        const auto num_elements = gv.size(0);
+        const int nt = omp_get_max_threads();
+        constexpr int max_chunk_size = 1000;
```

Comment:
Please add a comment why this is useful.

Reply:
Chunking in itself: Initialization of ElementContext is not free, so it is good to do it only once per chunk, and I think it is a good idea also for performance.

The choice of 1000 as size: Somewhat random. One could try to optimize this across a number of cases, but that is beyond my current ambition. Judging by the performance numbers it is at least reasonable for both SPE9 and spe10 (9k cells to 1M cells).

Reply:
i tried with maximum chunk sizes (ie nelem / nt with the remainder on the last thread) and it made no relevant difference.

Reply:
> and it made no relevant difference.

So should I change it, since there seems to be no benefit to smaller chunks?

Reply:
no, sorry i wasn't clear. it's better to have smaller chunks as it will in principle maximize threads in flight.

Reply:
I actually meant a comment directly in the code about this.